### DISCUSSION TITLE
Potentiometer Issue

### DISCUSSION THREAD

**Discussion by RasimUgurcan** [shanteacontrols/OpenDeck#175] [General]
Hello, I'm trying to make a prototype for a controller project. I simply built a prototype circuit. I am using blackpill411. I'm trying to understand the potentiometer connections first with a 4067. I installed the firmware following the guide. There is no problem in the operation of the device and its communication with the configurator. But I'm having a problem with the potentiometer. When I watched the activity, I saw that when turning the potentiometer, the potentiometer reacted only in a very short range between 0-127 values. In the middle of the potentiometer and in a very short range, it suddenly hovers between 0-127. What could be the problem?

Thank you.

**Comment by paradajz** [comment]
Can you post your YAML config?

**Reply by RasimUgurcan** [reply]
Frankly, I don't know where to find that file. I flashed the Blackpill via Windows as shown. I only have the .bin file. I'm pretty new, I know very little about software, I'm trying to achieve this by following your directions carefully. :) If you direct me, I can send the file to you. Forgive my ignorance.

**Reply by paradajz** [reply]
The firmware you have flashed doesn't support 4067 multiplexers as it uses native analog inputs. If you want to add multiplexer to your design, you need to use your own, custom config file. This is all explained here:
https://github.com/shanteacontrols/OpenDeck/wiki/Creating-custom-board-variant

**Reply by RasimUgurcan** [reply]
I am preparing a board over the schematics of the B variant with the Blackpill version. So I flashed opendeck3b.bin with variant B to Blackpill. I simply prepared the circuit according to the B variant of version 3. I made my connections by following the schematic and pcb examples. Shouldn't it be supported in this case? Everything is exactly the same. I'm not very good at these things. That's why I chose this way to make it easy. Otherwise, I may not be able to get out of this work.

**Reply by paradajz** [reply]
Well, how was I supposed to know that you've used `opendeck3b.bin`? You said you followed the guide, and the guide instructs you to use default firmware with standard analog inputs. Please provide all the details for your questions.

As for your problem, well, are you sure you have everything connected correctly?

**Reply by RasimUgurcan** [reply]
I chose this path because I was lacking in software. I did everything carefully about flashing. As for the board, I thought that I would have difficulties and even implement my project, so I went over the pcb. I put the prototype in the image below. If I could run the system with a single 4067 and a single potentiometer I would start drawing for a controls. But at the beginning I faced such a problem. Please be understanding. All of my connections were made as they were on the schematic. I fed the 4067 with 3v3. I placed all the S pins as in the diagram. I connected the potentiometer to one of the input pins of 4067 as in the diagram. The problem is that when I supply the potentiometer with 5V, the problem that I am telling you about occurs. In other words, it travels between 0-127 in a small range only in the middle part of the potentiometer. But when I feed the potentiometer with 3v3, the potentiometer seems to work correctly but after a value it gets random values. In this way, when I enter values ​​with Upper ADC offset, this situation seems to improve, but I give very high values ​​such as 80. And in this case, it moves from 0 to 127 in the first 30% cycle of the potentiometer. If I set Lower ADC 0 and Upper ADC 0, when the potentiometer is at the minimum position and the value is 9 when the value is at the maximum position, random numbers appear and at the end, random numbers stop and values ​​of 30 are displayed.
![WhatsApp Image 2022-09-27 at 23 26 50](https://user-images.githubusercontent.com/11924653/192628665-a1850363-b975-4620-a991-eee00fff4c6d.jpeg)
